source/common/network/socket_impl.cc

```diff
@@ -106,6 +106,7 @@ absl::optional<Address::IpVersion> SocketImpl::ipVersion() const {
     if (local_address_ != nullptr) {
       return local_address_->ip()->version();
     } else {
+      // TODO(fcoras) maybe use io_handle_->domain()?
```

Comment:
I agree, it makes sense to use domain() here. Either in this PR or a later one (up to you).

Reply:
Made the change and re-running tests locally. Will include the change in the next commit if nothing breaks.